### PR TITLE
Tpetra: Getting random pools correct

### PR DESCRIPTION
**Pull request by csiefer2**
This will avoid H2D transfers each time we call MultiVector::randomize(), like in Chebyshev initialization, say.

Issue: #12106 
Tests: Existing tests suffice
Stakeholder feedback: n/a

### CODE REVIEW

**Review by jhux2 — DISMISSED** [review]
lgtm

**Review by jhux2 — DISMISSED** [review]
This looks good to me.  Minor nitpick -- I might have either changed `isInitialized()` to `isReset()` or `resetPool` to `initializePool()` because the former is checking whether the latter has been called.

**Comment by jhux2** [comment]
> Would "isSet()" and "setPool()" be OK?

Sure, that's nicely consistent.

File: packages/tpetra/core/src/Tpetra_Details_Random.hpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+template<class ExecutionSpace>
+class Static_Random_XorShift64_Pool {
+public:
+  // The resetPool function will re-initialize the pool based on the system RNG and the MPI rank.
+  // On GPU architectures, this will likely involve non-trivial host-to-device transfers.
+  static void resetPool(int mpi_rank);
+
+  // The isInitialized function returns true if resetPool has been callled.
+  static bool isInitialized();
```
Comment (jhux2) [inline packages/tpetra/core/src/Tpetra_Details_Random.hpp:59]: Were you going to rename this guy?

**Review by jhux2 — APPROVED** [review]
lgtm

**Review by jhux2 — DISMISSED** [review]
Uh, this doesn't look good.

**Comment by jhux2** [comment]
@sebrowne Are the commits shown what is expected after the history rewrite?

**Comment by sebrowne** [comment]
My message left out the fact that a force push to the topic branch is required.  I'll send an update now.

**Comment by jhux2** [comment]
But the fact that there are 314 file changes is to be expected?  I see a lot of unrelated commits there.

**Comment by jhux2** [comment]
> Reload the page and there are only 6.

Doh!  That fixed it for me.

**Comment by sebrowne** [comment]
@csiefer2 thanks for testing this so quickly, glad to see it appears to work now.

**Review by jhux2 — APPROVED** [review]
doch

**Pull request merged**